source/common/buffer/buffer_impl.h

```diff
@@ -36,33 +36,35 @@ class Slice {
 public:
   using Reservation = RawSlice;
   using StoragePtr = std::unique_ptr<uint8_t[]>;
-
-  static constexpr uint32_t free_list_max_ = Buffer::Reservation::MAX_SLICES_;
-  using FreeListType = absl::InlinedVector<StoragePtr, free_list_max_>;
-  class FreeListReference {
-  private:
-    FreeListReference(FreeListType& free_list) : free_list_(free_list) {}
-    FreeListType& free_list_;
-    friend class Slice;
-  };
+  using SizedStorage = std::pair<uint64_t, StoragePtr>;
```

Comment:
Make this a struct for readability, so the accessors aren't `first` and `second` but `size_` and `ptr_` (or whatever names you choose.

Reply:
get it.